scripts/configure-installer-tests-cluster.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash
+# Setup to find nessasary data from cluster setup
+## Constants
+HTPASSWD_FILE="./htpass"
+USERPASS="developer"
+HTPASSWD_SECRET="htpasswd-secret"
+# Overrideable information
+DEFAULT_INSTALLER_ASSETS_DIR=${DEFAULT_INSTALLER_ASSETS_DIR:-$(pwd)}
+KUBEADMIN_USER=${KUBEADMIN_USER:-"kubeadmin"}
+KUBEADMIN_PASSWORD_FILE=${KUBEADMIN_PASSWORD_FILE:-"${DEFAULT_INSTALLER_ASSETS_DIR}/auth/kubeadmin-password"}
+# Exported to current env
+export KUBECONFIG=${KUBECONFIG:-"${DEFAULT_INSTALLER_ASSETS_DIR}/auth/kubeconfig"}
+
+# List of users to create
+USERS="developer odonoprojectattemptscreateproject odosingleprojectattemptscreate odologinnoproject odologinsingleproject1"
+
+# Check if nessasary files exist
+if [ ! -f $KUBEADMIN_PASSWORD_FILE ]; then
+    echo "Could not find kubeadmin password file"
+    exit 1
+fi
+
+if [ ! -f $KUBECONFIG ]; then
+    echo "Could not find kubeconfig file"
+    exit 1
+fi
+
+# Get kubeadmin password from file
+KUBEADMIN_PASSWORD=`cat $KUBEADMIN_PASSWORD_FILE`
+
+# Remove existing htpasswd file, if any
+if [ -f $HTPASSWD_FILE ]; then
+    rm -rf $HTPASSWD_FILE
+fi
+
+# Set so first time -c parameter gets applied to htpasswd
+HTPASSWD_CREATED=" -c "
+
+# Create htpasswd entries for all listed users
+for i in `echo $USERS`; do
+    htpasswd -b $HTPASSWD_CREATED $HTPASSWD_FILE $i $USERPASS
+    HTPASSWD_CREATED=""
+done
+
+# Login as admin user
+oc login -u $KUBEADMIN_USER -p $KUBEADMIN_PASSWORD
+
+# Create secret in cluster, removing if it already exists
+oc get secret $HTPASSWD_SECRET -n openshift-config &> /dev/null
+if [ $? -eq 0 ]; then
+    oc delete secret $HTPASSWD_SECRET -n openshift-config &> /dev/null
+fi
+oc create secret generic ${HTPASSWD_SECRET} --from-file=htpasswd=${HTPASSWD_FILE} -n openshift-config
+
+# Upload htpasswd as new login config
+oc apply -f - <<EOF
+apiVersion: config.openshift.io/v1
+kind: OAuth
+metadata:
+  name: cluster
+spec:
+  identityProviders:
+  - name: htpassidp
+    challenge: true
+    login: true
+    mappingMethod: claim
+    type: HTPasswd
+    htpasswd:
+      fileData:
+        name: ${HTPASSWD_SECRET}
+EOF
+
+# TODO : Find better way to check application of settings on cluster
+sleep 16
```

Comment:
With sleep of ```16s``` causes most of the time error of ```error: net/http: TLS handshake timeout``` on my aws cluster. After the oc apply my cluster taking bit longer time to be completely responsive for next oc command. 

Probably in prow 16s would be enough, but i would say to be in safer side use sleep time of atleast ```30s``` or more. sleep of ```30s``` works on my cluster on aws.

Reply:
Or something like 

Try until curl -f -k <end point>/healthz;do sleep 1;done

Reply:
That is something i want to do long term for now increasing time

Reply:
ok